src/shared/components/lazyMobXTable/LazyMobXTable.tsx

```diff
@@ -578,9 +582,23 @@ export default class LazyMobXTable<T> extends React.Component<LazyMobXTableProps
         showCountHeader: false
     };
 
-    public getDownloadData(): string
+    public getDownloadData(): Promise<string>
     {
-        return serializeData(this.store.downloadData);
+        return new Promise<string>((resolve, reject) => {
+            if (this.store.downloadStore) {
```

Comment:
We're going to get rid of nested "download" store.  Change some names to make this more clear.  That way it will make more sense why data doesn't come from store as promise return (b/c it's not in store!).  We have to leverage existing download functions to get it.

Reply:
can you explain more what's going on in this PR?